Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/Operations/Receive/HostObjectBuilder.cs

```diff
@@ -79,13 +93,31 @@ CancellationToken cancellationToken
         {
           string nestedLayerPath = $"{string.Join("\\", path)}";
           string datasetId = (string)_converter.Convert(obj);
-          conversionTracker[ctx] = new ObjectConversionTracker(obj, nestedLayerPath, datasetId);
+          conversionTracker[objectToConvert.TraversalContext] = new ObjectConversionTracker(
+            obj,
+            nestedLayerPath,
+            datasetId
+          );
         }
         else
         {
+          obj = _localToGlobalConverterUtils.TransformObjects(objectToConvert.AtomicObject, objectToConvert.Matrix);
+
           string nestedLayerPath = $"{string.Join("\\", path)}\\{obj.speckle_type.Split(".")[^1]}";
           Geometry converted = (Geometry)_converter.Convert(obj);
-          conversionTracker[ctx] = new ObjectConversionTracker(obj, nestedLayerPath, converted);
+          if (objectToConvert.Matrix.Count != 0)
+          {
+            // If it is transformed, we do not want to override 'conversionTracker' dict. Otherwise, we loose objects on bake.
+            conversionTracker[new TraversalContext(obj)] = new ObjectConversionTracker(obj, nestedLayerPath, converted);
```

Comment:
I think this is a bit tricky, because they are actually same objects.. can we assign is on the way like Brep_01, Brep_02, Brep_03... etc. If not, we should think some other way

Reply:
I actually tested the above line (with PropName as null though), and it doesn't seem to treat it as a same Traverse Context anymore, so all objects are still converted individually. And it gives later the proper ParentID to record as the dataset name. We don't actually care of the object Idshere, just the parent (because all objects will be recorded under one layer)

Reply:
If from this point on, the conversion tracker passes an object without TraversalContext, it won't be assigned to a Dataset with a Unique name. So, every following Layer with intances will create a dataset with the same generic name, overwritting the previous one.